packages/mobile/src/components/comments/CommentDrawer.tsx

```diff
@@ -122,23 +116,38 @@ export const CommentDrawer = () => {
         )}
         footerComponent={(props) => (
           <BottomSheetFooter {...props} bottomInset={insets.bottom}>
-            <CommentSectionProvider entityId={entityId}>
+            <CommentSectionProvider
+              entityId={entityId}
+              replyingToComment={replyingToComment}
+              setReplyingToComment={setReplyingToComment}
+              editingComment={editingComment}
+              setEditingComment={setEditingComment}
+            >
               <CommentDrawerForm />
             </CommentSectionProvider>
           </BottomSheetFooter>
         )}
         onDismiss={handleClose}
       >
-        <CommentSectionProvider entityId={entityId}>
+        <CommentSectionProvider
```

Comment:
so many providers 😵

Reply:
Yeah I really wish I could just wrap the whole drawer but unfortunately the content and footer end up in separate places in the hierarchy